### ISSUE TITLE
Taxonomy Changes: reload after applying changes

### ISSUE THREAD

**Issue by rovats** [finos/waltz#4706]
When changes are applied, the edit view isn't reloaded, and the *Apply Change* view isn't closed.

in `pending-taxonomy-changes-sub-section.js`:

```
function reload() {
        console.log("reload ?!")
    }
```

Also, `measurable-category-edit.html` isn't passing an `on-dismiss` handler, which is probably causing the apply change view to stay open:

```
<waltz-pending-taxonomy-changes-sub-section pending-changes="$ctrl.pendingChanges"
                                            on-apply-change="$ctrl.onApplyPendingChange"
                                            on-discard-change="$ctrl.onDiscardPendingChange">
</waltz-pending-taxonomy-changes-sub-section>
```

**Comment by davidwatkins73** [comment]
yeah - I _think_ there is an open task to finish this off.

**Comment by rovats** [comment]
OK, I will make these fixes at least as part of #4147, as I need these for a demo.
